problems/coj_problems.go

```diff
@@ -29,91 +54,86 @@ func generate_request_url(method, url_requested string, parameters map[string]st
     return request.URL.String(), nil
 }
 
-// COJJudgement is the struct to represent a COJ judment for a submition.
-type COJJudgement struct {
-    Id int
-    Date string
-    User string
-    Prob int
-    Judgment string
-    Errortestcase int
-    Time int
-    Memory string
-    Tam string
-    Lang string
-}
-
-// COJJudgmentResponse is the struct to represent a collection of COJ Judgments.
-type COJJudgmentResponse struct {
-    Collection [] COJJudgement
-}
-
+// createNetClient generates a http client to make request.
+// 
+// It returns a pointer of type http.Client to make request to servers.
+// We do the request in this way so we can control the timeout, otherwise we may have troubles
+// if the server we are requesting for fails.
+func createNetClient() (*http.Client){
+    const timeoutRequest = 10 // Requests timeouts
+    const timeoutDial = 180  // limits the time spent establishing a TCP connection, often around 3 minutes.
+    const timeoutTSL = 30    // limits the time spent performing the TLS handshake.
+
+    // cap the TCP connect and TLS handshake timeouts
+    // as well as establishing an end-to-end request timeout.
+    netTransport := &http.Transport{
+      Dial: (&net.Dialer{
+        Timeout: timeoutDial * time.Second,
+      }).Dial,
+      TLSHandshakeTimeout: timeoutTSL * time.Second,
+    }
+    netClient := &http.Client{
+      Timeout: time.Second * timeoutRequest,
+      Transport: netTransport,
+    }
 
-// SubmitionsProblemInfo represents the important information of the submitions to a problem
-// The values for has_solved && has_tried are "YES" && "NO"
-type SubmitionsProblemInfo struct {
-    has_solved  string
-    has_tried   string
+    return netClient
 }
 
-
 // getHasSolvedInCOJ returns the important information that said if a user "USERNAME" has solve or has tried
 // the problem "PID" in the COJ.
 func getHasSolvedInCOJ(pid string, username string) (SubmitionsProblemInfo, error){
-    submitionsProblemInfo := SubmitionsProblemInfo{"NO", "NO"}
+    sumbitionsInfo := SubmitionsProblemInfo{"NO", "NO"}
 
     // Parameters sent to the API
-    api_parameters := map[string]string{ "username": username, "pid": pid};
+    apiParameters := map[string]string{ "username": username, "pid": pid};
     // Generate the URL for make the request
-    api_url, url_err := generate_request_url("GET", "http://coj.uci.cu/api/judgment", api_parameters)
+    apiURL, errApiURL := generateRequestURL("GET", "http://coj.uci.cu/api/judgment", apiParameters)
```

Comment:
I think `err` would be just fine here, there  are no other errors around, up to you.